connect/src/protocols/tokenBridge/tokenTransfer.ts

```diff
@@ -627,14 +644,26 @@ export namespace TokenTransfer {
     dstChain: ChainContext<N, Chain>,
     transfer: Omit<TokenTransferDetails, "from" | "to">,
   ): Promise<TransferQuote> {
-    const srcDecimals = await srcChain.getDecimals(transfer.token.address);
+    const srcTb = await srcChain.getTokenBridge();
+    let srcToken: NativeAddress<Chain>;
+    if (isNative(transfer.token.address)) {
+      srcToken = await srcTb.getWrappedNative();
+    } else if (UniversalAddress.instanceof(transfer.token.address)) {
+      try {
+        srcToken = (await srcTb.getWrappedAsset(transfer.token)) as NativeAddress<Chain>;
+      } catch (e) {
+        srcToken = await srcTb.getTokenNativeAddress(srcChain.chain, transfer.token.address);
```

Comment:
I noticed in some other places that we don't check the exception, but I'm going to fix that. We should check that the exception being thrown is what we expect otherwise re-raise it.